### DISCUSSION TITLE
Passing Parameters of No Specific Type to Fluents

### DISCUSSION THREAD

**Discussion by Yomaster10** [aiplan4eu/unified-planning#274] [Q&A]
Is there any way to create fluents that can receive a parameter that doesn't have a specific type (i.e. it's just of type `object`, which is possible in PDDL)?

**Comment by Framba-Luca** [comment]
Currently no, we don't offer this feature.

**Reply by mikand** [reply]
@Framba-Luca that is not exact: you can simply create the type `object` yourself and derive every other type from it. At this point, a parameter of type `object` accepts everything. 

@Framba-Luca can you construct and post an example here?

**Reply by Framba-Luca** [reply]
Ah yes! I got the question wrong; I thought @Yomaster10 wanted an automatic default.

In the following example, besides showing what is requested, I will also show a counterexample!

It's important to note that this works only for `UserTypes`!
```
from unified_planning.shortcuts import *

# Types definition
object_type = UserType("object") 
type_1 = UserType(name="type_1", father=object_type) # NOTE: object_type is set as father
type_2 = UserType("type_2", object_type)
type_3 = UserType("type_3") # NOTE: type_3 does not derive from object_type!

# objects definition
obj_1 = Object("obj_1", type_1)
obj_2 = Object("obj_2", type_2)
obj_3 = Object("obj_3", type_3)

# fluent definition and possible uses!
fluent = Fluent("fluent", BoolType(), param_name=object_type) #NOTE: The first and only parameters has type "object"
fluent(obj_1) # Correct expression!
fluent(obj_2) # Correct expression!
fluent(obj_3) # Wrong expression, raises an UPTypeError!
```
